networkx/algorithms/community/modularity_max.py

```diff
@@ -14,7 +14,7 @@
 ]
 
 
-def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1):
+def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1, enforce_n_communities=False):
```

Comment:
These two keyword arguments `n_communities` and `enforce_n_communities` seem like there ought to be a better naming scheme.  The `n_communities` is a cutoff after which we stop. The `enforce_n_communities` is a signal to keep going until n is reached.  I think what I don't like is that `n_communities` means two different things depending on what another keyword arg's value is.

I'd like to propose that we use `best_n_communities=None` to be number n when we want exactly n communities.
And `n_communities` continue to be the cutoff number n of communities at which to stop if the optimum has not been reached.  A better term might be `cutoff`.  That would require a deprecation, so I'm fine either way.  I'm open to other suggestions too...  

Thoughts?

Reply:
yup, sounds more consistent than the current naming scheme. Will use `cutoff` and `best_n_communities`.